akka-actor/src/main/scala/akka/io/TcpConnection.scala

```diff
@@ -326,7 +326,7 @@ private[io] abstract class TcpConnection(val tcp: TcpExt, val channel: SocketCha
       }
     }
 
-  def abort(): Unit = {
+  def abort(info: Option[ConnectionInfo]): Unit = {
```

Comment:
I think it's a bit paradox that the static bits about the connection, the `ConnectionInfo`, are passed around everywhere explicitly, while the actual mutable state lives in a `var`. Maybe let's make this a `var` as well and remove all the passing around?